### PR TITLE
Modern diag_manager: Store the current_model_time in the diag_object

### PR DESCRIPTION
**Pull request by uramirez8707**
**Description**
Adds a variable to the diag_object to store the current model time (which gets updated using the time_step in diag_send_complete) + updates tests

Fixes # (issue)

**How Has This Been Tested?**
CI

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

**Review by thomas-robinson — APPROVED** [review]

File: diag_manager/fms_diag_file_object.F90 [hunk 1]
```diff
@@ -745,9 +745,12 @@ end subroutine add_axes
 !> @brief adds the start time to the fileobj
 !! @note This should be called from the register field calls. It can be called multiple times (one for each variable)
 !! So it needs to make sure that the start_time is the same for each variable. The initial value is the base_time
-subroutine add_start_time(this, start_time)
-  class(fmsDiagFile_type), intent(inout)       :: this            !< The file object
+subroutine add_start_time(this, start_time, model_time)
+  class(fmsDiagFile_type), intent(inout)       :: this           !< The file object
   TYPE(time_type),         intent(in)          :: start_time     !< Start time to add to the fileobj
+  TYPE(time_type),         intent(out)         :: model_time     !< The current model time
```
Comment (thomas-robinson) [inline diag_manager/fms_diag_file_object.F90:751]: This is he start time.  Where does it become the current model time?

File: diag_manager/fms_diag_object.F90 [hunk 2]
```diff
@@ -226,7 +229,7 @@ integer function fms_register_diag_field_obj &
      call fileptr%set_file_domain(fieldptr%get_domain(), fieldptr%get_type_of_domain())
      call fileptr%init_diurnal_axis(this%diag_axis, this%registered_axis, diag_field_indices(i))
      call fileptr%add_axes(axes, this%diag_axis, this%registered_axis, diag_field_indices(i))
-     call fileptr%add_start_time(init_time)
+     call fileptr%add_start_time(init_time, this%current_model_time)
```
Comment (thomas-robinson) [inline diag_manager/fms_diag_object.F90:232]: Here it's the current model time.  In the routine, it is the start time.

**Review by thomas-robinson — CHANGES_REQUESTED** [review]
You need to update the description of what `model_time` is in `add_start_time`.  It'e the start time of the model.  You should probably have a setter for `this%current_model_time` instead.

File: diag_manager/fms_diag_file_object.F90 (hunk 1 above)
Comment (uramirez8707) [reply diag_manager/fms_diag_file_object.F90:751]: This is being called from the `register_fields` calls (the beginning of the run)
so the `current_model_time` is the `start_time`

File: diag_manager/fms_diag_object.F90 (hunk 2 above)
Comment (uramirez8707) [reply diag_manager/fms_diag_object.F90:232]: This is in the initialization, the current model time is the start_time.

File: diag_manager/fms_diag_file_object.F90 (hunk 1 above)
Comment (uramirez8707) [reply diag_manager/fms_diag_file_object.F90:751]: it then gets updated after every diag_send_complete

**Pull request merged**